codec-http2/src/main/java/io/netty/handler/codec/http2/Http2DataMessage.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http2;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufHolder;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+import io.netty.util.IllegalReferenceCountException;
+
+public class Http2DataMessage extends Http2StreamMessage implements ByteBufHolder {
+    private final ByteBuf content;
+    private final boolean endStream;
+    private final int padding;
+
+    /**
+     * Equivalent to {@code new Http2DataMessage(content, false)}.
+     *
+     * @param content non-{@code null} payload
+     */
+    public Http2DataMessage(ByteBuf content) {
+        this(content, false);
+    }
+
+    /**
+     * Equivalent to {@code new Http2DataMessage(Unpooled.buffer(0, 0), endStream)}.
+     *
+     * @param endStream whether this data should terminate the stream
+     */
+    public Http2DataMessage(boolean endStream) {
+        this(Unpooled.buffer(0, 0), endStream);
+    }
+
+    /**
+     * Equivalent to {@code new Http2DataMessage(content, endStream, 0)}.
+     *
+     * @param content non-{@code null} payload
+     * @param endStream whether this data should terminate the stream
+     */
+    public Http2DataMessage(ByteBuf content, boolean endStream) {
+        this(content, endStream, 0);
+    }
+
+    /**
+     * Construct a new data message.
+     *
+     * @param content non-{@code null} payload
+     * @param endStream whether this data should terminate the stream
+     * @param padding additional bytes that should be added to obscure the true content size
+     */
+    public Http2DataMessage(ByteBuf content, boolean endStream, int padding) {
+        this.content = checkNotNull(content, "content");
+        this.endStream = endStream;
+        if (padding < 0) {
```

Comment:
Padding is an unsigned byte https://tools.ietf.org/html/draft-ietf-httpbis-http2-17#section-6.1